webapp/tests/karma/ts/components/search-bar/search-bar.component.spec.ts

```diff
@@ -152,4 +200,171 @@ describe('Search Bar Component', () => {
     tick();
     expect(component.showClearIcon()).to.be.false;
   }));
+
+  describe('Barcode scanner support', () => {
+    it('should return true if BarcodeDetector is supported, user has permission and is not admin', async () => {
+      sessionService.isAdmin.returns(false);
+      authService.has.resolves(true);
+      browserDetectorService.isDesktopUserAgent.returns(false);
+      sinon.resetHistory();
+
+      await component.ngAfterViewInit();
+
+      expect(component.isBarcodeScannerAvailable).to.be.true;
+      expect(sessionService.isAdmin.calledOnce).to.be.true;
+      expect(browserDetectorService.isDesktopUserAgent.called).to.be.true;
+      expect(authService.has.calledOnce).to.be.true;
+      expect(authService.has.args[0]).to.have.members([ CAN_USE_BARCODE_SCANNER ]);
+    });
+
+    it('should return false if browser is desktop', async () => {
+      sessionService.isAdmin.returns(false);
+      authService.has.resolves(true);
+      browserDetectorService.isDesktopUserAgent.returns(true);
+      translateService.instant.returns('some text');
+      const setSnackbarContentSpy = sinon.spy(GlobalActions.prototype, 'setSnackbarContent');
+      sinon.resetHistory();
+
+      await component.ngAfterViewInit();
+
+      expect(component.isBarcodeScannerAvailable).to.be.false;
+      expect(sessionService.isAdmin.calledOnce).to.be.true;
+      expect(browserDetectorService.isDesktopUserAgent.called).to.be.true;
+      expect(authService.has.calledOnce).to.be.true;
+      expect(authService.has.args[0]).to.have.members([ CAN_USE_BARCODE_SCANNER ]);
+      expect(translateService.instant.calledWith('barcode_scanner.warning.not_supported')).to.be.true;
+      expect(feedbackService.submit.calledWith('some text')).to.be.true;
+      expect(setSnackbarContentSpy.calledWith('some text')).to.be.true;
+      expect(telemetryService.record.calledWith('search_by_barcode:not_supported'));
+    });
+
+    it('should return false if user does not have permission', async () => {
+      sessionService.isAdmin.returns(false);
+      authService.has.resolves(false);
+      sinon.resetHistory();
+
+      await component.ngAfterViewInit();
+
+      expect(component.isBarcodeScannerAvailable).to.be.false;
+      expect(sessionService.isAdmin.calledOnce).to.be.true;
+      expect(authService.has.calledOnce).to.be.true;
```

Comment:
And if you agree with the previous one, also add here: `expect(browserDetectorService.isDesktopUserAgent.called).to.be.true;`